tests/lib/bin/mountinfo-tool

```diff
@@ -911,34 +913,44 @@ class RenumberMountOptionsTests(unittest.TestCase):
         """
         numbers are allocated from subsets matching the key, this reduces delta.
         """
-        self.assertEqual(renumber_mount_option("size=100", self.seen), "size=0")
-        self.assertEqual(renumber_mount_option("size=100", self.seen), "size=0")
-        self.assertEqual(renumber_mount_option("size=200", self.seen), "size=1")
-        self.assertEqual(renumber_mount_option("size=100", self.seen), "size=0")
+        self.assertEqual(renumber_mount_option("pipe_ino=100", self.seen), "pipe_ino=0")
+        self.assertEqual(renumber_mount_option("pipe_ino=100", self.seen), "pipe_ino=0")
+        self.assertEqual(renumber_mount_option("pipe_ino=200", self.seen), "pipe_ino=1")
+        self.assertEqual(renumber_mount_option("pipe_ino=100", self.seen), "pipe_ino=0")
         self.assertEqual(renumber_mount_option("fd=21", self.seen), "fd=0")
         self.assertEqual(renumber_mount_option("fd=21", self.seen), "fd=0")
         self.assertEqual(renumber_mount_option("fd=45", self.seen), "fd=1")
         self.assertEqual(renumber_mount_option("fd=21", self.seen), "fd=0")
 
+    def test_renumber_size_variable(self):
+        # type: () -> None
+        """
+        size is special-cased and always rewritten to the same value because it is prone to fluctuations
+        """
+        self.assertEqual(renumber_mount_option("size=100", self.seen), "size=VARIABLE")
+        self.assertEqual(renumber_mount_option("size=200", self.seen), "size=VARIABLE")
+
     def test_renumber_devtmpfs_opts(self):
         # type: () -> None
         """
         certain devtmpfs options are renumbered.
 
             23 98 0:6 / /dev rw,nosuid shared:21 - devtmpfs devtmpfs rw,size=4057388k,nr_inodes=1014347,mode=755
 
-        Here the size and nr_inodes options are not deterministic and need to be rewritten.
+        Here the size and nr_inodes options are not deterministic and need to
+        be rewritten.  The size quantity is very susceptible to free memory
+        fluctuations and is treated specially.
         """
         # Options size= and nr_inodes= are renumbered.
-        self.assertEqual(renumber_mount_option("size=4057388k", self.seen), "size=0")
+        self.assertEqual(
+            renumber_mount_option("size=4057388k", self.seen), "size=VARIABLE"
+        )
         self.assertEqual(
             renumber_mount_option("nr_inodes=1014347", self.seen), "nr_inodes=0"
         )
         # Option mode= is not renumbered.
         self.assertEqual(renumber_mount_option("mode=755", self.seen), "mode=755")
-        self.assertEqual(
-            self.seen, {("size", "4057388k"): 0, ("nr_inodes", "1014347"): 0}
-        )
+        self.assertEqual(self.seen, {("nr_inodes", "1014347"): 0})
```

Comment:
While at it, is `nr_inodes` stable across reboots?

Reply:
Yeah, I haven't seen any issues with that. I think it depends on storage size, if anything.

Reply:
Digging into the kernel, looks like it's calculated here: https://elixir.bootlin.com/linux/latest/source/mm/shmem.c#L116 so indeed stable.